File: backend/src/main/java/com/koda/platform/platform/configuration/application/CompanySettings.java
package com.koda.platform.platform.configuration.application;

import com.koda.platform.shared.domain.tenant.TenantId;
import java.time.Instant;
import java.util.UUID;

public record CompanySettings(
    UUID id,
    TenantId tenantId,
    String commercialName,
    String legalName,
    String countryCode,
    String defaultLocale,
    String defaultCurrency,
    String timeZone,
    String logoUrl,
    String faviconUrl,
    String loginImageUrl,
    String primaryColor,
    String secondaryColor,
    String themeMode,
    String dateFormat,
    String timeFormat,
    String numberLocale,
    String currencyFormat,
    long version,
    Instant updatedAt
) {
}

File: frontend/src/platform/configuration/companySettings.ts
import type {
  CompanyProfileBranding,
  CompanyProfileRegional,
  CompanyProfileTenant,
  CompanyRuntimeProfile,
  ThemeMode,
} from './companyProfile';

export interface CompanySettingsTenant extends CompanyProfileTenant {
  legalName: string;
}

export interface CompanySettingsBranding extends CompanyProfileBranding {
  themeMode: ThemeMode;
}

export type CompanySettingsRegional = CompanyProfileRegional;

export interface CompanySettings {
  id: string;
  tenant: CompanySettingsTenant;
  branding: CompanySettingsBranding;
  regional: CompanySettingsRegional;
  updatedAt: string;
  version: number;
}

export interface UpdateCompanySettingsRequest {
  dateFormat: string;
  defaultCurrency: string;
  defaultLocale: string;
  faviconUrl: string | null;
  currencyFormat: string;
  logoUrl: string | null;
  loginImageUrl: string | null;
  numberLocale: string;
  primaryColor: string;
  secondaryColor: string | null;
  themeMode: ThemeMode;
  timeFormat: string;
  timeZone: string;
  version: number;
}

export function companySettingsToRuntimeProfile(settings: CompanySettings): CompanyRuntimeProfile {
  return {
    tenant: {
      id: settings.tenant.id,
      commercialName: settings.tenant.commercialName,
      countryCode: settings.tenant.countryCode,
    },
    branding: settings.branding,
    regional: settings.regional,
    updatedAt: settings.updatedAt,
  };
}
